examples/extra_element_integer/random_cluster.cpp

```diff
@@ -0,0 +1,148 @@
+#include "libmesh/dof_map.h"
+#include "libmesh/elem.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/exodusII_io.h"
+#include "libmesh/libmesh.h"
+#include "libmesh/linear_implicit_system.h"
+#include "libmesh/mesh.h"
+#include "libmesh/mesh_generation.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/system.h"
+
+#include <cstdlib>
+#include <ctime>
+#include <random>
+#include <vector>
+
+using namespace libMesh;
+
+void CreateMesh(libMesh::Mesh &mesh, int nx, int ny) {
+  libMesh::MeshTools::Generation::build_square(mesh, nx, ny, 0.0, 10.0, 0.0,
+                                               10.0);
+}
+
+void CreateMesh(libMesh::Mesh &mesh, int nx, int ny, int nz) {
+  libMesh::MeshTools::Generation::build_cube(mesh, nx, ny, nz, 0.0, 10.0, 0.0,
+                                             10.0, 0.0, 10.0);
+}
+
+void PopulateRandomIntegers(libMesh::Mesh &mesh, LinearImplicitSystem &system) {
+  DofMap &dof_map = system.get_dof_map();
+  for (const auto &elem : mesh.element_ptr_range()) {
+
+    std::vector<dof_id_type> dof_indices;
+    dof_map.dof_indices(elem, dof_indices);
+    int random_int = rand() % 5;
+    system.solution->set(dof_indices[0], 5 - random_int);
+  }
+}
+
+void AddVaribalesToSystem(libMesh::EquationSystems &equation_systems,
+                          LinearImplicitSystem &system,
+                          std::string var_name, bool needs_re_init =false) {
+    system.add_variable(var_name, CONSTANT, MONOMIAL);
+    if (!needs_re_init){
+        equation_systems.init();
+    }
+    else{
+        equation_systems.reinit();
+    }
+}
+
+void FindCluster(libMesh::Mesh &mesh, libMesh::DofMap &dof_map,
+                 libMesh::DofMap &local_dof_map, LinearImplicitSystem &system,
+                 LinearImplicitSystem &local_system,
+                 const std::string &variable_name,
+                 libMesh::EquationSystems &equation_systems,
+                 const unsigned int index) {
```

Comment:
I think you can simplify this to:
```suggestion
unsigned int FindCluster(libMesh::Mesh &mesh,  LinearImplicitSystem &system,
                 const std::string &variable_name) {
```
and
* generate the `DofMap` inside this function
* don't do anything with the local_system yet - it doesn't need to exist yet - it's only purpose is to record the cluster_id for saving to file
* name the new elem_integer with the same name as the `variable_name`
* return the index of the new elem_integer